sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotInstantiateSharedClasses.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.ComponentModel.Composition;
+
+namespace Tests.Diagnostics
+{
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    class SharedClass {}
+
+    [PartCreationPolicy(CreationPolicy.NonShared)]
+    class NonSharedClass { }
+
+    [PartCreationPolicy(CreationPolicy.Any)]
+    class AnyClass { }
+
```

Comment:
How easy is it to test the "ignore if test" case?

Reply:
Do we have a rule warns the user about a missing [Export] attribute? (doesn't make sense to have a _PartCreationPolicy_ attribute on a type that isn't exported).

Reply:
I was actually wondering the same and couldn't find anything (yet) [regarding the `ExportAttribute`]

Reply:
RSPEC-4159 is kind of related, but we probably want a separate rule. Feel free to create.

"Ignore if test" is easy, I'll add a test.

Reply:
Not really related. I will suggest the rule on dogfood-rule.